modules/pubwiseAnalyticsAdapter.js

```diff
@@ -54,72 +73,268 @@ function enrichWithUTM(dataBag) {
+        localStorage.setItem(setNamespace(prop), utmKeys[prop]);
       }
     }
   } catch (e) {
-    utils.logInfo(`${analyticsName}Error`, e);
+    pwInfo(`Error`, e);
     dataBag['error_utm'] = 1;
   }
   return dataBag;
 }
 
-function sendEvent(eventType, data) {
-  utils.logInfo(`${analyticsName}Event ${eventType} ${pwAnalyticsEnabled}`, data);
+function expireUtmData() {
+  pwInfo(`Session Expiring UTM Data`);
+  for (let prop in utmKeys) {
+    localStorage.removeItem(setNamespace(prop));
+  }
+}
+
+function enrichWithCustomSegments(dataBag) {
+  // c_site: '', c_script_type: '', c_slot1: '', c_slot2: '', c_slot3: '', c_slot4: ''
+  if (configOptions.custom) {
+    if (configOptions.custom.c_script_type) {
+      dataBag['c_script_type'] = configOptions.custom.c_script_type;
+    }
+
+    if (configOptions.custom.c_host) {
+      dataBag['c_host'] = configOptions.custom.c_host;
+    }
+
+    if (configOptions.custom.c_slot1) {
+      dataBag['c_slot1'] = configOptions.custom.c_slot1;
+    }
+
+    if (configOptions.custom.c_slot2) {
+      dataBag['c_slot2'] = configOptions.custom.c_slot2;
+    }
+
+    if (configOptions.custom.c_slot3) {
+      dataBag['c_slot3'] = configOptions.custom.c_slot3;
+    }
+
+    if (configOptions.custom.c_slot4) {
+      dataBag['c_slot4'] = configOptions.custom.c_slot4;
+    }
+  }
+
+  return dataBag;
+}
+
+function setNamespace(itemText) {
+  return pwNamespace.concat('_' + itemText);
+}
+
+function localStorageSessTimeoutName() {
+  return setNamespace(sessTimeoutName);
+}
+
+function localStorageSessName() {
+  return setNamespace(sessName);
+}
+
+function extendUserSessionTimeout() {
+  localStorage.setItem(localStorageSessTimeoutName(), Date.now().toString());
+}
+
+function userSessionID() {
+  return localStorage.getItem(localStorageSessName()) ? localStorage.getItem(localStorageSessName()) : '';
+}
+
+function sessionExpired() {
+  let sessLastTime = localStorage.getItem(localStorageSessTimeoutName());
+  return (Date.now() - parseInt(sessLastTime)) > sessTimeout;
+}
+
+function flushEvents() {
+  if (pwEvents.length > 0) {
+    let dataBag = {metaData: metaData, eventList: pwEvents.splice(0)}; // put all the events together with the metadata and send
+    ajax(configOptions.endpoint, (result) => pwInfo(`Result`, result), JSON.stringify(dataBag));
+  }
+}
+
+function isIngestedEvent(eventType) {
+  const ingested = [
+    CONSTANTS.EVENTS.AUCTION_INIT,
+    CONSTANTS.EVENTS.BID_REQUESTED,
+    CONSTANTS.EVENTS.BID_RESPONSE,
+    CONSTANTS.EVENTS.BID_WON,
+    CONSTANTS.EVENTS.BID_TIMEOUT,
+    CONSTANTS.EVENTS.AD_RENDER_FAILED,
+    CONSTANTS.EVENTS.TCF2_ENFORCEMENT
+  ];
+  return ingested.indexOf(eventType) !== -1;
+}
+
+function markEnabled() {
+  pwInfo(`Enabled`, configOptions);
+  pwAnalyticsEnabled = true;
+  setInterval(flushEvents, 100);
+}
+
+function pwInfo(info, context) {
+  utils.logInfo(`${analyticsName} ` + info, context);
+}
+
+/*
```

Comment:
Consider removing unused code; it can be resumed from git history.

Reply:
Removed